core/src/main/java/org/apache/calcite/schema/Schema.java

```diff
@@ -71,6 +75,8 @@ public interface Schema {
    */
   Set<String> getTableNames();
 
+  @Nullable Table getTable(String name, boolean caseSensitive);
```

Comment:
I agree. Adding `caseSensitive` should be an orthogonal change.

And I'm not even convinced we want to make that change.

Reply:
> Why we add `caseSensitive` here? I do not get what is the relationship between it and `lazy`
1、@Nullable Table getTable(String name, boolean caseSensitive); This method  only be called when your SQL references a certain table .
2、  @Nullable Table getTable(String name) can ignore case sentitive ,because  the program internally processed the case sentitive in org.apache.calcite.jdbc.CachingCalciteSchema#getImplicitTable of org.apache.calcite.util.NameSet#range by org.apache.calcite.schema.impl.AbstractSchema#getTableNames provide all table names. 

However ,  @Nullable Table getTable(String name, boolean caseSensitive) don't relies on getTableNames  method ,so  i add caseSensitive param.